python/triton/runtime/jit.py

```diff
@@ -77,6 +80,10 @@ def is_triton_builtin(func):
             key = func_cache_key + noinline
             self.hasher.update(key.encode("utf-8"))
 
+    def visit_FunctionDef(self, node):
+        # Save the local name which may hide the global name.
+        self.local_name = [arg.arg for arg in node.args.args]
```

Comment:
I think it is incomplete because we can also define a local variable with assignment, not as function arguments.

For example:

```Python
@triton.jit
def kernel(...):
    ptr = tl.load(...)
    print(ptr.dtype.element_ty)

ptr = torch.tensor(...)
```

Reply:
@Sarbojit2019 Do you plan to address the issue in this PR?

Reply:
Yes, proposed the fix in latest commit.